apis/kueue/v1beta1/admissioncheck_types.go

```diff
@@ -25,16 +25,14 @@ type CheckState string
 const (
 	// CheckStateRetry means that the check cannot pass at this moment, back off (possibly
 	// allowing other to try, unblock quota) and retry.
-	// A workload having at least one check in the state,
-	// will be evicted if admitted will not be considered
-	// for admission.
+	// A workload having at least one check in the state, will be evicted if admitted and
+	// not be considered for admission.
 	CheckStateRetry CheckState = "Retry"
 
 	// CheckStateRejected means that the check will not pass in the near future. It is not worth
 	// to retry.
-	// NOTE: The admission behaviour is currently the same as for retry,
-	// we can consider marking the workload as "Finished" with a failure
-	// description.
+	// A workload having at least one check in the state, will be evicted if admitted will and
```

Comment:
```suggestion
	// A workload having at least one check in the state is marked finished. An admitted workload
	// is evicted before marked as finished.
```

Reply:
A suggestion, I think it will be easier to follow.

Reply:
The order should also matter

Reply:
I didn't get - "order of what"? If my suggestion is wrong in some way, could you please try to propose another update to the comment to clarify.

Reply:
Of operations, the wl is first evicted after that is marked as finished.

Reply:
Got it, maybe then 
```
// A workload having at least one check in the state is marked finished. An admitted workload
// is also evicted.
```

Reply:
Can you please propose an edit, because I'm not sure what you mean.

Reply:
I did already.